app-modules/student-data-model/src/Http/Controllers/Api/V1/Students/StudentPrograms/ListStudentProgramsController.php

```diff
@@ -0,0 +1,128 @@
+    #[QueryParameter('filter[cum_gpa]', description: 'Filter the results where the program\'s Cumulative GPA contains the provided number.', type: 'float')]
+    #[QueryParameter('filter[semester]', description: 'Filter the results where the program\'s Semester contains the provided string.', type: 'string')]
+    #[QueryParameter('filter[descr]', description: 'Filter the results where the program\'s Description contains the provided string.', type: 'string')]
+    #[QueryParameter('filter[foi]', description: 'Filter the results where the program\'s Field of Interest contains the provided string.', type: 'string')]
+    #[QueryParameter('filter[change_dt]', description: 'Filter the results where the program\'s Change Date matches the provided date using comparison operators. Supported: =, <, <=, >, >=. Format: YYYY-MM-DD H:i:s.', type: 'datetime')]
+    #[QueryParameter('filter[declare_dt]', description: 'Filter the results where the program\'s Declare Date matches the provided date using comparison operators. Supported: =, <, <=, >, >=. Format: YYYY-MM-DD H:i:s.', type: 'datetime')]
+    #[QueryParameter('page[number]', description: 'Control which page of programs is returned in the response.', type: 'int', default: 1)]
+    #[QueryParameter('page[size]', description: 'Control how many programs are returned in the response.', type: 'int', default: 30)]
+    #[QueryParameter('sort', description: 'Control the order of programs that are returned in the response. Ascending order is used by default, prepend the sort with `-` to sort descending.', type: 'string', default: 'sisid', examples: [
+        'acad_career' => new Example('acad_career'),
+        'division' => new Example('division'),
+        'acad_plan' => new Example('acad_plan'),
+        'prog_status' => new Example('prog_status'),
+        'cum_gpa' => new Example('cum_gpa'),
+        'semester' => new Example('semester'),
+        'descr' => new Example('descr'),
+        'foi' => new Example('foi'),
+        'change_dt' => new Example('change_dt'),
+        'declare_dt' => new Example('declare_dt'),
+    ])]
+    public function __invoke(Request $request, Student $student): AnonymousResourceCollection
+    {
+        Gate::authorize('viewAny', Student::class);
+        Gate::authorize('view', $student);
+        Gate::authorize('viewAny', Program::class);
+
+        return QueryBuilder::for($student->programs())
+            ->allowedFilters([
+                AllowedFilter::partial('sisid'),
+                AllowedFilter::partial('acad_career'),
+                AllowedFilter::partial('division'),
+                AllowedFilter::callback('acad_plan', function (Builder $query, string $value) {
+                    $query->whereRaw('acad_plan::text ILIKE ?', ["%{$value}%"]);
+                }),
+                AllowedFilter::partial('prog_status'),
+                AllowedFilter::operator('cum_gpa', FilterOperator::DYNAMIC),
```

Comment:
Oh it looks like this is already a dynamic operator, so it looks like you just need to update the documentation messaging that I commented on above.